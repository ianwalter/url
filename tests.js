const { test } = require('@ianwalter/bff')
const createUrl = require('.')

test('simple url', t => {
  const urlString = 'https://ianwalter.dev/'
  const url = createUrl(urlString)
  t.expect(url.toString()).toBe(urlString)
  t.expect(url.hostname).toBe('ianwalter.dev')
  t.expect(url.origin).toBe('https://ianwalter.dev')
  t.expect(JSON.parse(JSON.stringify(url))).toMatchSnapshot()
})

test('url with query', async t => {
  const urlString = 'https://ianwalter.dev/q?product=ipad&product=ipod&page=1'
  const url = createUrl(urlString)
  t.expect(url.toString()).toBe(urlString)
  t.expect(JSON.parse(JSON.stringify(url))).toMatchSnapshot()
})

test('url with base', t => {
  const path = '/spotlight-on-decision-tree'
  const url = createUrl('https://ianwalter.dev', path)
  t.expect(url.toString()).toBe(`https://ianwalter.dev${path}`)
  t.expect(url.pathname).toBe(path)
})

test('url from object', t => {
  const base = createUrl('https://ianwalter.dev')
  const url = createUrl(base, '/spotlight-on-decision-tree#aboutAuthor')
  t.expect(JSON.parse(JSON.stringify(url))).toMatchSnapshot()
})
