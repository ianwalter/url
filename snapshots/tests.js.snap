// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple url 1`] = `
Object {
  "hash": "",
  "host": "ianwalter.dev",
  "hostname": "ianwalter.dev",
  "href": "https://ianwalter.dev/",
  "origin": "https://ianwalter.dev",
  "password": "",
  "pathname": "/",
  "port": "",
  "protocol": "https:",
  "search": Object {},
  "username": "",
}
`;

exports[`url from object 1`] = `
Object {
  "hash": "#aboutAuthor",
  "host": "ianwalter.dev",
  "hostname": "ianwalter.dev",
  "href": "https://ianwalter.dev/spotlight-on-decision-tree#aboutAuthor",
  "origin": "https://ianwalter.dev",
  "password": "",
  "pathname": "/spotlight-on-decision-tree",
  "port": "",
  "protocol": "https:",
  "search": Object {},
  "username": "",
}
`;

exports[`url with query 1`] = `
Object {
  "hash": "",
  "host": "ianwalter.dev",
  "hostname": "ianwalter.dev",
  "href": "https://ianwalter.dev/q?product=ipad&product=ipod&page=1",
  "origin": "https://ianwalter.dev",
  "password": "",
  "pathname": "/q",
  "port": "",
  "protocol": "https:",
  "search": Object {
    "page": "1",
    "product": Array [
      "ipad",
      "ipod",
    ],
  },
  "username": "",
}
`;
